### PR TITLE
Hyper-V support & related fixes

### PR DESCRIPTION
**Pull request by mnowaksuse**
Requires https://github.com/os-autoinst/os-autoinst/pull/798.

CaaSP 1.0: http://assam.suse.cz/tests/92
SLES 12 SP3 JeOS: http://assam.suse.cz/tests/94
SLES: http://assam.suse.cz/tests/99
SLES w/ UEFI: http://assam.suse.cz/tests/113

### CODE REVIEW

File: lib/bootloader_setup.pm [hunk 1]
```diff
@@ -148,7 +146,7 @@ sub bootmenu_default_params {
             assert_screen check_var('UEFI', 1) ? 'inst-video-typed-grub2' : 'inst-video-typed', 4;
         }
 
-        if (!get_var("NICEVIDEO")) {
+        if (!get_var("NICEVIDEO") && !is_jeos && !is_casp) {
             if (is_jeos) {
```
Comment (asmorodskyi) [inline lib/bootloader_setup.pm]: Some wrong , or in line 149 or in line 150

File: lib/bootloader_setup.pm [hunk 2]
```diff
@@ -319,6 +324,61 @@ sub specific_bootmenu_params {
...
+    for (1 .. 4) { send_key "down"; }
+    send_key "end";
+
+    if (get_var("NETBOOT")) {
+        type_string_slow " install=" . get_netboot_mirror;
+        save_screenshot();
+    }
+    send_key "spc";
+
+    # if(get_var("PROMO")) {
```
Comment (asmorodskyi) [inline lib/bootloader_setup.pm]: remove commented code

File: tests/installation/bootloader_hyperv.pm [hunk 3]
```diff
@@ -11,112 +11,121 @@
...
-
-        $t->open(get_var('HYPERV_SERVER'));
-        $t->login(get_var('HYPERV_USERNAME'), get_var('HYPERV_PASSWORD'));
-
-        # This is the expected shell prompt: '$ ' otherwise cmd() will timeout.
-        # * in Bash: PS1='$ ' in ~/.bash_profile
-        # * as a CMD.exe: set user var PROMPT to $$$S
-        $t->prompt('/\$ /i');
+sub is_hyperv_gen2_vm() {
+    return get_var('UEFI');
```
Comment (asmorodskyi) [inline tests/installation/bootloader_hyperv.pm]: Why not just call get_var('UEFI') where you need it ?

File: tests/jeos/mount_by_label.pm [hunk 4]
```diff
@@ -18,11 +18,17 @@ use testapi;
 sub run() {
     # Valid mounts are by text(proc,mem), label, partlabel. Invalid mounts are by UUID, PARTUUID,
     # and path (note that /dev/disk/by-(part)label/ is considered 'as a (part)label mount', just
-    # in different format).
-    my $invalid = script_output "! grep -e '^/dev/disk/by-[^\\(label,partlabel\\)]' -e '^UUID' -e '^PARTUUID' /etc/fstab";
+    # in different format). Except for Hyper-V where we use UUID by design.
```
Comment (sysrich) [inline tests/jeos/mount_by_label.pm]: Do we use UUID by design, or does the product us UUID?

File: lib/bootloader_setup.pm (hunk 2 above)
Comment (mnowaksuse) [inline lib/bootloader_setup.pm]: I did not created it the commented code, I just drag it from on place to another.

**Comment by mnowaksuse** [comment]
I believe it's ready now.

**Comment by okurz** [comment]
the hyperv powershell magic is impressive :-)

**Pull request merged**